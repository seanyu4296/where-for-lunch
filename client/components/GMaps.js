import React from 'react';
import PropTypes from 'prop-types';
import { compose, withProps } from 'recompose';
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker,
} from 'react-google-maps';

const GMaps = ({ lat, lng }) => (
  <GoogleMap defaultZoom={15} defaultCenter={{ lat, lng }}>
    <Marker position={{ lat, lng }} />
  </GoogleMap>
);

GMaps.propTypes = {
  lat: PropTypes.number.isRequired,
  lng: PropTypes.number.isRequired,
};

export default compose(
  withProps({
    googleMapURL:
      'https://maps.googleapis.com/maps/api/js?key=AIzaSyAgtZVw8UTy-bO0b6WkF5LktS1_c4DzyPA&v=3.exp&libraries=geometry,drawing,places',
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `100%` }} />,
    mapElement: <div style={{ height: `100%` }} />,
  }),
  withScriptjs,
  withGoogleMap,
)(GMaps);
