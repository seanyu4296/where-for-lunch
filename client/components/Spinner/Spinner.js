import React from 'react';
import styles from './Spinner.css';

export default () => (
  <div className={styles.root}>
  <div className={styles['lds-ring']}>
    <div />
    <div />
    <div />
    <div />
  </div>
  </div>
);
