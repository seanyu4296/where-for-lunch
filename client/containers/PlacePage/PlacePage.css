@import '../../styles/variables.css';
.root {
  width: 100%;
  box-shadow: 0 0 5px #d0d0d0;
  min-height: 100vh;
  background-color: #fafafa;
  border-radius: 3px;
}
.carouselImage {
  object-fit: contain;
  width: 100%;
  height: 400px;
  background: black;
}

.coverImage {
  object-fit: cover;
  width: 100%;
}

.topContent {
  display: flex;
  padding: 0 1rem;
  margin-bottom: 1rem;
  align-items: center;
}
.headerContainer {
  flex: 1;
}

.rightHeaderContainer {
  padding: 0 1rem;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.ratingContainer {
  display: flex;
  border-radius: 3px;
  align-items: center;
  padding: 0.3rem;
  margin-bottom: 0.2rem;
}

.lowRating {
  background-color: #e04a3f;
}

.midRating {
  background-color: #f5ca1d;
}

.highRating {
  background-color: #24a252;
}

.openOrClosed {
  border-radius: 3px;
  padding: 0.5rem;
  margin: 0;
  margin-top: 0.5rem;
}

.closed {
  background-color: gainsboro;
}

.rating {
  color: white;
  font-size: 2rem;
}
.ratingDenominator {
  color: white;
  opacity: 0.5;
  font-size: 1.8rem;
}

.open {
  background-color: #099e44;
}

.title {
  font-family: $body-font;
  font-size: 2rem;
  color: #33373d;
  margin: 0;
  padding: 0;
  margin-bottom: 0.5rem;
}

.subtitle {
  font-size: 1rem;
  display: block;
  font-family: $body-font;
  color: #89959b;
}

.contentContainer {
  padding: 1.5rem 1rem;
}

.categoryContainer {
  color: rgba(0, 0, 0, 0.87);
  border: none;
  height: 32px;
  cursor: default;
  padding: 0;
  display: inline-flex;
  outline: none;
  font-size: 0.8125rem;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
    box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  white-space: nowrap;
  align-items: center;
  font-family: $body-font;
  border-radius: 16px;
  text-decoration: none;
  justify-content: center;
  background-color: #e0e0e0;
  margin-right: 0.5rem;
}

.category {
  margin: 0.5rem;
}

.divider {
  border-top: 1px solid #e7e7e7;
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
}

.yelpButton {
  background-color: #33aedc;
  cursor: pointer;
  display: inline-block;
  min-height: 1em;
  outline: 0;
  border: none;
  vertical-align: baseline;
  color: rgba(0, 0, 0, 0.6);
  font-family: $body-font;
  margin: 1rem 0.25rem 0 0;
  padding: 0.78571429em 1.5em;
  text-transform: none;
  text-shadow: none;
  line-height: 1em;
  font-style: normal;
  text-align: center;
  text-decoration: none;
  border-radius: 0.28571429rem;
  box-shadow: 0 0 0 1px transparent inset, 0 0 0 0 #e7e7e7 inset;
  -webkit-user-select: none;
}

.divider {
  border-top: 1px solid #e7e7e7;
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
}
.yelpText {
  padding: 1rem;
  color: #fff;
  font-size: 1rem;
}

.detailsContainer {
  padding: 1rem;
  width: 100%;
  display: flex;
}

.mainDetailsContainer {
  flex: 1;
  padding-right: 1rem;
}

.mapContainer {
  flex-basis: 48%;
  background-color: gainsboro;
  height: 500px;
}
.detailsHeader {
  font-family: $body-font;
  font-size: 1.5rem;
  color: #33373d;
  border-bottom: 2px solid #cb202d;
  padding-bottom: 0.5rem;
  margin-bottom: 1rem;
}
.mainDetailsContainer {
  display: flex;
  flex-direction: column;
}

.detail {
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
}
.key {
  font-size: 1rem;
  margin-bottom: 5px;
  font-weight: bold;
  color: black;
}

.value {
  font-size: 1rem;
  color: black;
  margin-bottom: 4px;
}
