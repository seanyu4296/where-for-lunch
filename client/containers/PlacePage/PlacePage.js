import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Carousel from 'nuka-carousel';
import { bindActionCreators } from 'redux';
import styles from './PlacePage.css';
import { fetchPlaceDetailsRequest } from '../../actions/placeActions';
// const getHours = () => {};

class PlacePage extends React.Component {
  componentDidMount() {
    this.props.actions.fetchPlaceDetailsRequest();
  }

  render() {
    const {
      place: { data },
    } = this.props;
    const {
      /*       name,
      image_url,
      is_closed,
      url: yelpUrl,
      phone,
      display_phone,
      review_count,
      categories = [],
      rating,
      location,
      coordinates, */
      photos = [],
      /*       price,
      hours = [], */
    } =
      data || {};
    return (
      <div className={styles.root}>
        {photos.length ? (
          <Carousel>
            {photos.map((v, i) => {
              return <img className={styles.carouselImage} key={i} src={v} />;
            })}
          </Carousel>
        ) : null}
      </div>
    );
  }
}

PlacePage.propTypes = {
  place: PropTypes.shape({
    data: PropTypes.object,
  }),
  actions: PropTypes.shape({
    fetchPlaceDetailsRequest: PropTypes.func,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
};

PlacePage.defaultProps = {
  place: {
    data: {},
  },
};

const mapStateToProps = ({ place }) => ({ place });
const mapDispatchToProps = (
  dispatch,
  {
    match: {
      params: { id },
    },
  },
) => ({
  actions: bindActionCreators(
    {
      fetchPlaceDetailsRequest: () => fetchPlaceDetailsRequest(id),
    },
    dispatch,
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlacePage);
