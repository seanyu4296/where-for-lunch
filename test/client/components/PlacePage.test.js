import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Carousel from 'nuka-carousel';
import { mount } from 'enzyme';
import { PlacePage } from '../../../client/containers/PlacePage/PlacePage';
import Spinner from '../../../client/components/Spinner/Spinner';
import GMaps from '../../../client/components/GMaps';

describe('With Enzyme, PlacePage Component', () => {
  const baseProps = {
    actions: { fetchPlaceDetailsRequest: () => {} },
    match: { params: { id: 'test' } },
  };
  test('should have one Spinner when place fetching is true', () => {
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <PlacePage {...baseProps} place={{ fetching: true, data: {} }} />
      </MemoryRouter>,
    );
    expect(wrapper.find(Spinner).length).toBe(1);
  });
  test('should have image Carousel present when there are multiple photos', () => {
    const wrapper = mount(
      <PlacePage
        {...baseProps}
        place={{ fetching: false, data: { photos: ['p', 'h'] } }}
      />,
    );
    expect(wrapper.find(Carousel).exists()).toEqual(true);
  });
  test('should have GMaps present when lat lng is present', () => {
    const wrapper = mount(
      <PlacePage
        {...baseProps}
        place={{
          fetching: false,
          data: { coordinates: { latitude: 1, longitude: 2 } },
        }}
      />,
    );
    expect(wrapper.find(GMaps).exists()).toEqual(true);
  });
  test('should not have GMaps when missing lat lng data', () => {
    const wrapper = mount(
      <PlacePage
        {...baseProps}
        place={{
          fetching: false,
          data: {},
        }}
      />,
    );
    expect(wrapper.find(GMaps).exists()).toEqual(false);
    expect(wrapper.contains(<span>No Location Available</span>)).toEqual(true);
  });
});

test('With Jest Snapshot, Place Page renders content when place data is present', () => {
  const place = {
    data: {
      id: 'TEST',
      name: 'TEST',
      is_closed: false,
      url: 'https://www.test.com/test',
      phone: '+6300000',
      display_phone: '+63 0000000',
      review_count: 2,
      categories: [
        {
          alias: 'TEST',
          title: 'TEST',
        },
      ],
      rating: 4.5,
      location: {
        address1: '',
        address2: '',
        address3: '',
        city: '',
        zip_code: '',
        country: '',
        state: '',
        display_address: [''],
        cross_streets: '',
      },
      coordinates: {
        latitude: 1,
        longitude: 1,
      },
      photos: [],
      hours: [
        {
          open: [
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 0,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 1,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 2,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 3,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 4,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 5,
            },
            {
              is_overnight: true,
              start: '1100',
              end: '0200',
              day: 6,
            },
          ],
          hours_type: 'REGULAR',
          is_open_now: true,
        },
      ],
      transactions: [],
    },
  };
  const props = {
    place,
    actions: { fetchPlaceDetailsRequest: () => {} },
    match: { params: { id: 'test' } },
  };
  const placePage = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <PlacePage {...props} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(placePage).toMatchSnapshot();
});
