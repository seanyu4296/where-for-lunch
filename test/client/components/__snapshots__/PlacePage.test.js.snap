// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest Snapshot, Place Page renders content when place data is present 1`] = `
<div
  className="root"
>
  <div
    className="contentContainer"
  >
    <div
      className="topContent"
    >
      <div
        className="headerContainer"
      >
        <h3
          className="title"
        >
          TEST
        </h3>
        <span
          className="subtitle"
        >
          
           Â·
           
          <div
            className="categoryContainer"
          >
            <span
              className="category"
            >
              TEST
            </span>
          </div>
        </span>
        <a
          className="yelpButton"
          href="https://www.test.com/test"
        >
          <span
            className="yelpText"
          >
            View in Yelp
          </span>
        </a>
      </div>
      <div
        className="rightHeaderContainer"
      >
        <div
          className="ratingContainer lowRating midRating highRating"
        >
          <span
            className="rating"
          >
            4.5
          </span>
          <span
            className="ratingDenominator"
          >
            /5
          </span>
        </div>
        <p
          className="openOrClosed open"
        >
          OPEN
        </p>
      </div>
    </div>
    <div
      className="divider"
    />
    <div
      className="detailsContainer"
    >
      <div
        className="mainDetailsContainer"
      >
        <h4
          className="detailsHeader"
        >
          Details
        </h4>
        <div
          className="detail"
        >
          <p
            className="key"
          >
            Phone
          </p>
          <p
            className="value"
          >
            +63 0000000
          </p>
        </div>
        <div
          className="detail"
        >
          <p
            className="key"
          >
            Address
          </p>
          <p
            className="value"
          >
            
          </p>
        </div>
        <div
          className="detail"
        >
          <p
            className="key"
          >
            Review Count in Yelp
          </p>
          <p
            className="value"
          >
            2
          </p>
        </div>
        <div
          className="detail"
        >
          <p
            className="key"
          >
            Opening Hours
          </p>
          <p
            className="value"
          >
            Sunday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Monday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Tuesday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Wednesday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Thursday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Friday
            : 
            1100 - 0200
          </p>
          <p
            className="value"
          >
            Saturday
            : 
            1100 - 0200
          </p>
        </div>
      </div>
      <div
        className="mapContainer"
      />
    </div>
  </div>
</div>
`;
